Fix package install instructions in "Numerical Sage" tutorial (#37313)
sagemath/sage: src/doc/en/thematic_tutorials/numerical_sage/mpi4py.rst

    print("hello world")

>     print("my rank is: %d"%comm.rank)

Reviewer: It is a good idea to update the syntax, i.e.
```suggestion
    print(f"my rank is: {comm.rank}")
```

To run it you can do (from the command line in your Sage
directory)

.. CODE-BLOCK:: shell-session

    mpirun -np 5 ./sage -python mpi_1.py

The command ``mpirun -np 5`` starts 5 copies of a program under MPI. In

> this case we have 5 copies of Sage in pure Python mode run the

Reviewer: ```suggestion
this case we have 5 copies of Sage in pure Python mode running the
```

    print("I received this:")
    print(data)

The same command as above with ``mpi_1.py`` replaced by ``mpi_2.py`` will
produce 5 outputs and you will see each process creates an array and

> then passes it to the next guy (where the last guy passes to the

Reviewer: ```suggestion
produce 5 outputs. Each process will create an array and pass
it to the next process, where the last process passes to the
```

> first.) Note that ``MPI.size`` is the total number of MPI

Reviewer: ```suggestion
first. Note that ``MPI.size`` is the total number of MPI
```

waiting to receive, etc. This will not lock regardless of how large

> of an array we pass.

Reviewer: I can't seem to comment on L108-109, but replace `he` -> `it`. My suggestion:

```diff
- Now the first process initiates a send, and then process 1 will be
- ready to receive and then he will send and process 2 will be
- waiting to receive, etc. This will not lock regardless of how large
- of an array we pass.
+ Now, process 0 sends the data to process 1, then waits to receive from process ``MPI.size - 1``.
+ Simultaneously, process 1 will send the data to process 2, then receives the data from process 0.
+ This will not lock even if the array transmitted is huge.
```

(Please line wrap for me)

A common idiom is to have one process, usually the one with rank 0,
act as a leader. That process sends data out to the other

> processes and processes the results and decides how further

Reviewer: ```suggestion
processes, compute on the results, and decides how much further
```

    print("I got this array:")
    print(v)

The ``scatter`` command takes a list and evenly divides it amongst all
the processes. Here the root process creates a matrix (which is
viewed as a list of rows) and then scatters it to everybody (root's
``sendbuf`` is divided equally amongst the processes). Each process
prints the row it got. Note that the ``scatter`` command is executed by
everyone, but when root executes it, it acts as a ``send`` and a
``receive`` (root gets one row from itself), while for everyone else it
is just a ``receive``.

There is a complementary ``gather`` command that collects results from
all the processes into a list. The next example uses ``scatter`` and
``gather`` together. Now the root process scatters the rows of a
matrix, each process then squares the elements of the row it gets.
Then the rows are all gathered up again by the root process who

> collects them into a new matrix.

Reviewer: ```suggestion
``gather`` together. Now the root process scatters the rows of a
matrix. Each process squares the elements of the row it receives.
The root process then gathers the rows into a new matrix.
```
